### PR TITLE
[PLANNER-1382] Don't create accessors for primitive setters

### PR DESCRIPTION
**Pull request by rsynek**

### CODE REVIEW

**Comment by rsynek** [comment]
@ge0ffrey finally the SonarCloud integration seems to work - I needed to delete and recreate the project in SonarCloud. Maybe the permission update and/or reinstallation of the SonarCloud app on github severed the connection somehow.

File: optaplanner-core/src/test/java/org/optaplanner/core/impl/testdata/domain/shadow/cyclic/invalid/TestdataCyclicReferencedShadowedEntity.java [hunk 1]
```diff
@@ -43,7 +43,7 @@ public static GenuineVariableDescriptor buildVariableDescriptorForValue() {
     }
 
     private TestdataValue value;
-    private boolean barber;
+    private Boolean barber;
```
Comment (ge0ffrey) [inline optaplanner-core/src/test/java/org/optaplanner/core/impl/testdata/domain/shadow/cyclic/invalid/TestdataCyclicReferencedShadowedEntity.java]: Any reason to revert this part? Doesn't it just reduce test coverage?

**Comment by rsynek** [comment]
@ge0ffrey fixed even for the shadow variable

**Comment by ge0ffrey** [comment]
Looks perfect now, I'll add something in the docs that we do support primitives on shadow vars.

**Pull request merged**